### PR TITLE
Sample which showcases how to connect traces and logs with OpenTelemetry

### PR DESCRIPTION
**Pull request by lailabougria**

### CODE REVIEW

File: samples/open-telemetry/logging/Core_8/GenericHost/Program.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+{
+    public static void Main(string[] args)
+    {
+        CreateHostBuilder(args).Build().Run();
+    }
+
+    private static IHostBuilder CreateHostBuilder(string[] args)
+    {
+        var builder = Host.CreateDefaultBuilder(args);
+        builder.UseWindowsService();
```
Comment (timbussmann) [inline samples/open-telemetry/logging/Core_8/GenericHost/Program.cs:23]: do we use windows service for our samples?

File: samples/open-telemetry/logging/Core_8/GenericHost/Program.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+        #region generic-host-nservicebus
+
+        builder.UseNServiceBus(ctx =>
+        {
+            var endpointConfiguration = new EndpointConfiguration("Samples.Hosting.GenericHost");
+            endpointConfiguration.UseTransport(new LearningTransport());
+
+            endpointConfiguration.DefineCriticalErrorAction(OnCriticalError);
```
Comment (timbussmann) [inline samples/open-telemetry/logging/Core_8/GenericHost/Program.cs:32]: is that required/useful for this sample?

File: samples/open-telemetry/logging/Core_8/GenericHost/Program.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        #endregion
+
+        #region otel-config
+
+        builder.ConfigureServices(services =>
+        {
+            services.AddOpenTelemetryTracing(config => config
+                                                       .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService("MyEndpointName"))
+                                                       .AddSource("NServiceBus.Diagnostics")
```
Comment (timbussmann) [inline samples/open-telemetry/logging/Core_8/GenericHost/Program.cs:45]:
```suggestion
                                                       .AddSource("NServiceBus.*")
```

File: samples/open-telemetry/logging/Core_8/GenericHost/Program.cs (hunk 1 above)
Comment (lailabougria) [reply samples/open-telemetry/logging/Core_8/GenericHost/Program.cs:23]: I based this off another sample which was using a windows service, I can easily change it

File: samples/open-telemetry/logging/Core_8/GenericHost/Program.cs (hunk 2 above)
Comment (lailabougria) [reply samples/open-telemetry/logging/Core_8/GenericHost/Program.cs:32]: Not specifically for the sample, but it seems like a good enough practice to not remove it? As stated before, I based this off another sample

File: samples/open-telemetry/logging/sample.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+summary: How to connect OpenTelemetry traces to logs
+reviewed: 2022-06-27
+component: Extensions.Logging
+related:
+- nservicebus/logging
+---
+
+This sample shows how to configure an NServiceBus endpoint to collect telemetry data and link the traces and logs together.
+
+First, OpenTelemetry needs to be configured on the endpoint:
```
Comment (mikeminutillo) [inline samples/open-telemetry/logging/sample.md:12]: Something about this felt off. I'm not happy with the suggestion either

```suggestion
First, OpenTelemetry needs to be configured to capture NServiceBus traces:
```

**Review by mikeminutillo — APPROVED** [review]

**Review by timbussmann — APPROVED** [review]

File: samples/open-telemetry/logging/Core_8/GenericHost/MyMessageHandler.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    public MyMessageHandler(ILogger<MyMessageHandler> logger)
+    {
+        this.logger = logger;
+    }
+
+    public Task Handle(MyMessage message, IMessageHandlerContext context)
+    {
+        #region log-statement
+
+        logger.LogInformation($"Received message #{message.Number}");
```
Comment (bbrandt) [inline samples/open-telemetry/logging/Core_8/GenericHost/MyMessageHandler.cs:18]: String interpolation should not be used when logging.

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2254

https://www.youtube.com/watch?v=6zoMd_FwSwQ

Comment (ramonsmits) [reply samples/open-telemetry/logging/Core_8/GenericHost/MyMessageHandler.cs:18]: agreed, should be using structured logging and unfortunately we are a serious violator: https://github.com/Particular/NServiceBus/issues/6136

Comment (ramonsmits) [reply samples/open-telemetry/logging/Core_8/GenericHost/MyMessageHandler.cs:18]: Applied your feedback in https://github.com/Particular/docs.particular.net/commit/d472f003fd6214942d5dc66e1d148bee9bf32d8b for this specific sample.

**Pull request merged**